framework/branch-key-store.md

```diff
@@ -74,29 +88,72 @@ A list of AWS KMS [grant tokens](https://docs.aws.amazon.com/kms/latest/develope
 
 The DynamoDb Client used to put and get keys from the backing DDB table.
 
-If not provided, one will be created in the region of the supplied KMS Key ARN.
+If the AWS KMS Configuration is KMS Key ARN or KMS MRKey ARN,
+and no DynamoDb Client is provided,
+a new DynamoDb Client MUST be created
+with the region of the supplied KMS ARN.
+
+If the AWS KMS Configuration is Discovery,
+and no DynamoDb Client is provided,
+a new DynamoDb Client MUST be created
+with the default configuration.
+
+If the AWS KMS Configuration is MRDiscovery,
+and no DynamoDb Client is provided,
+a new DynamoDb Client MUST be created
+with the region configured in the MRDiscovery.
 
 ### KMS Client
 
 The KMS Client used when wrapping and unwrapping keys.
 
-If not provided, one will be created in the region of the supplied KMS Key ARN.
+If the AWS KMS Configuration is KMS Key ARN or KMS MRKey ARN,
+and no KMS Client is provided,
+a new KMS Client MUST be created
+with the region of the supplied KMS ARN.
+
+If the AWS KMS Configuration is Discovery,
+and no KMS Client is provided,
+a new KMS Client MUST be created
+with the default configuration.
+
+If the AWS KMS Configuration is MRDiscovery,
+and no KMS Client is provided,
+a new KMS Client MUST be created
+with the region configured in the MRDiscovery.
 
-On initialization the KeyStore MUST append a user agent string to the AWS KMS SDK Client with the
-value `aws-kms-hierarchy`.
+On initialization the KeyStore SHOULD
```

Comment:
Why MUST to SHOULD?

Reply:
Because we have never done it and have no backlog item to do so.

Also, customers really MUST have the right to be excluded from metrics.